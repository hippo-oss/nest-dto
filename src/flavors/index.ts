export * as openapi from './openapi';
export * as strict from './strict';
