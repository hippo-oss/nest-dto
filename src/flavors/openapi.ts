import { composeDecoratorFactories } from '@hippo-oss/dto-decorators';
import { CLASS_DECORATORS } from '@hippo-oss/class-decorators';
import { OPENAPI_DECORATORS } from '@hippo-oss/openapi-decorators';

export const DECORATORS = composeDecoratorFactories([
    CLASS_DECORATORS,
    OPENAPI_DECORATORS,
]);

export const {
    IsBoolean,
    IsDate,
    IsDateString,
    IsEnum,
    IsInteger,
    IsNested,
    IsNumber,
    IsString,
    IsUUID,
} = DECORATORS;
