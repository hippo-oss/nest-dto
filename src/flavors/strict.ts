import { composeDecoratorFactories } from '@hippo-oss/dto-decorators';
import { CLASS_DECORATORS } from '@hippo-oss/class-decorators';

export const DECORATORS = composeDecoratorFactories([
    CLASS_DECORATORS,
]);

export const {
    IsBoolean,
    IsDate,
    IsDateString,
    IsEnum,
    IsInteger,
    IsNested,
    IsNumber,
    IsString,
    IsUUID,
} = DECORATORS;
