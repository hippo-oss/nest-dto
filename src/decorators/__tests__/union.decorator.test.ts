import { IsNumber, IsString } from '../../flavors/openapi';
import {
    createOpenAPIDocument,
    createOpenAPIFixtureModule,
} from '../../testing';
import { Union } from '../union.decorator';

class Foo {

    @Union([
        IsString,
        IsNumber,
    ])
    bar!: string | number;
}

describe('decorators', () => {
    describe('Union', () => {
        it('generates a oneOf definition', async () => {
            const FixtureModule = createOpenAPIFixtureModule(Foo);
            const document = await createOpenAPIDocument(FixtureModule);

            expect(document).toMatchSnapshot();
        });
    });
});
