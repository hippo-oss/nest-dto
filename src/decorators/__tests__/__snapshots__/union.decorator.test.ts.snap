// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decorators Union generates a oneOf definition 1`] = `
Object {
  "components": Object {
    "schemas": Object {
      "Foo": Object {
        "properties": Object {
          "bar": Object {
            "oneOf": Array [
              Object {
                "type": "string",
              },
              Object {
                "type": "number",
              },
            ],
          },
        },
        "required": Array [
          "bar",
        ],
        "type": "object",
      },
    },
  },
  "info": Object {
    "contact": Object {},
    "description": "",
    "title": "",
    "version": "1.0.0",
  },
  "openapi": "3.0.0",
  "paths": Object {
    "/": Object {
      "get": Object {
        "operationId": "FixtureController_fixture",
        "parameters": Array [],
        "responses": Object {
          "default": Object {
            "content": Object {
              "application/json": Object {
                "schema": Object {
                  "$ref": "#/components/schemas/Foo",
                },
              },
            },
            "description": "",
          },
        },
      },
    },
  },
  "servers": Array [],
  "tags": Array [],
}
`;
